libs/shared-web/src/hooks/useAuthController.ts

```diff
@@ -122,19 +134,22 @@ export function useAuthController() {
   const logout = () => {
     setTokenCallback(null);
     queryClient.clear();
-    navigate('/auth/login');
+    navigate(ROUTES.AUTH_LOGIN);
     segment.reset();
   };
 
   return {
     isLoggedIn,
-    user,
-    isUserLoading,
+    user: {
+      ...user,
+      organizationId: jwtPayload?.organizationId,
+      environmentId: jwtPayload?.environmentId,
+    } satisfies IUserWithContext,
+    isUserLoading: isUserLoading && isLoggedIn,
```

Comment:
Loading state is now additionally determined by whether the user is logged in. A user cannot be loading if they are not logged in.